### PR TITLE
Added S3 setup for secure production storage

### LINKED ISSUE #23: Deployment - Add S3 container for production mode
**User Story:**

As a user I want the system to have a secure and reliable way to store and access both datasets and models in production, so I can trust that my predictions are accurate and available whenever I need them.

**Persona:**

Users who depend on stable and consistent model performance for their tasks. Ken for example, a CTO of a startup, would want to keep his data secure and in a controlled environment, safe from bad-actors and technological issues (like an outage).

**Feature:**

- Configure an S3 container specifically for production environments.
- Ensure secure access and seamless integration with the production server.
- Implement best practices for S3 usage, such as versioning, access control, and encryption.

**Business Value:**

Setting up an S3 container for production mode will ensure that ML models can access necessary datasets and resources reliably and securely. This will enhance the robustness of the platform by providing consistent data storage and retrieval for production environments.

**Tasks:**

1. Research and configure an S3 container for the production environment.
2. Set up access control policies and encryption for secure data storage.
3. Integrate the S3 container with the production server to load models and datasets.
4. Test the configuration to ensure data can be loaded from S3 without issues.
5. Document the setup process, including configuration and integration steps.

**Acceptance Criteria:**

- The S3 container is set up and configured for production.
- Access control policies and encryption are correctly implemented.
- The production server can securely access and load models/data from the S3 container.
- Detailed documentation is available for future setup and troubleshooting.

**Acceptance Tests:**

- Confirm that the production server can successfully load models/data from the S3 container.
- Verify that access to the S3 container is secure, with proper access control and encryption.
- Test multiple scenarios to ensure data is retrieved smoothly without interruptions.

### PR DESCRIPTION
**Pull request by zsabri91**
Features:

- Set up S3 storage specifically for production to securely store datasets and models
- Implement access control policies and encryption for secure and reliable data handling
- Integrated S3 with the production server to enable seamless loading of datasets and models

Testing:

- Deploy to production environment
- Verify that the production server can upload, access, and load datasets/models from S3
- Confirm access control and encryption policies are enforced on S3
- Test multiple scenarios to ensure smooth, secure data retrieval in production

Issue: #23

### CODE REVIEW

File: mlite_rails_app/app/models/user.rb [hunk 1]
```diff
@@ -12,6 +12,8 @@ class User < ApplicationRecord
 
   private
 
+  # has_one_attached :profile_picture
+
```
Comment (Mohammad4844) [inline mlite_rails_app/app/models/user.rb]: remove this

File: mlite_rails_app/config/environments/development.rb [hunk 2]
```diff
@@ -31,8 +31,7 @@
     config.cache_store = :null_store
   end
 
-  # Store uploaded files on the local file system (see config/storage.yml for options).
-  config.active_storage.service = :local
+  config.active_storage.service = :amazon
```
Comment (Mohammad4844) [inline mlite_rails_app/config/environments/development.rb]: put :local

File: mlite_rails_app/config/storage.yml [hunk 3]
```diff
@@ -32,3 +32,13 @@ local:
...
+
+amazon:
+  service: S3
+  access_key_id: <%= ENV['AWS_ACCESS_KEY_ID'] %>
+  secret_access_key: <%= ENV['AWS_SECRET_ACCESS_KEY'] %>
+  region: us-east-1
+  bucket: <%= ENV['AWS_S3_BUCKET'] %>
+#  upload:
+#    cache_control: "public, max-age=31536000"
+
```
Comment (Mohammad4844) [inline mlite_rails_app/config/storage.yml]: remove

File: mlite_rails_app/config/routes.rb
Comment (Mohammad4844) [inline mlite_rails_app/config/routes.rb:1]: remove

File: mlite_rails_app/config/initializers/aws.rb
Comment (Mohammad4844) [inline mlite_rails_app/config/initializers/aws.rb:1]: remove this too

**Review by Mohammad4844 — CHANGES_REQUESTED** [review]
most of it looks good and works as expected. just a few redundant files to get rid of.

**Review by Mohammad4844 — APPROVED** [review]

**Comment by sjiang1** [comment]
Grading Feedback:

Quality of Code: Good
Participation in Code Review: Good
Post-Review Changes: Good

Deduction: Minor (-2 points)
Reason: Some commit messages do not adhere to the required format specified in the lecture. Please ensure future commit messages align with the established guidelines.

**Pull request merged**